subsys/bluetooth/services/ots/ots_client.c

```diff
@@ -1156,7 +1196,70 @@ static int oacp_read(struct bt_conn *conn,
 
 	return err;
 }
+static int oacp_write(struct bt_conn *conn,
+		     struct bt_otc_internal_instance_t *inst,
+			 uint8_t *buf, size_t len, off_t offset, uint8_t mode)
+{
+	int err;
+	struct bt_gatt_ots_l2cap *l2cap;
+
+	LOG_DBG("");
+	if (!inst->otc_inst->oacp_handle) {
+		BT_DBG("Handle not set");
+		return -EINVAL;
+	} else if (inst->busy) {
+		return -EBUSY;
+	} else if (cur_inst) {
+		return -EBUSY;
+	}
+
+	/* TODO: How do we ensure that the L2CAP is connected in time for the
+	 * transfer?
+	 */
```

Comment:
I guess there are 2 ways of achieving this:
1) Store the parameters, call bt_gatt_ots_l2cap_connect and on the `connected` callback we either abort or continue the procedure
2) Like 1), but where we use a semaphore/mutex similar to `bt_hci_cmd_send_sync` to avoid storing the parameters

Reply:
This part is copied from oacp_read().
Maybe we should also invite Asbjørn to discuss.

Reply:
Ping @asbjornsabo @kapi-no @ekohandel Ideas?